src/pages/tag.js

```diff
@@ -54,6 +54,13 @@ export default class TagShop extends Component {
     if (this.props.catalogItems !== prevProps.catalogItems) {
       this.trackEvent(this.props);
     }
+
+    if (this.props.tag !== prevProps.tag) {
```

Comment:
I wonder if this could be moved to `getDerivedStateFromProps` to avoid the initial render before the redirect. Currently, the tag page will start to render and then it redirects to it self without the params, which leads to not the best UX.